### ISSUE TITLE
Reinstall under same server_name not possible?

### ISSUE THREAD

**Issue by wuast94** [matrix-org/synapse#10424]
I had recently a db crash and had to make clean install.

now i have problem to recreate my chats with other becouse in a federation network i cannot simply do a fresh install with the same server_name. 

But on the other side i cant change ther server name becouse i want to use the one i have. 

is there any way to get this fixed? if not there should be a way in future to fix this problem :)

**Comment by DoM1niC** [comment]
Remove and recreate DB works fine and will init from itself

**Comment by wuast94** [comment]
Yes it does and that isn't the problem...the problem is that I can't start a chat again with a friend (we had a chat and groups before reinstall).

**Comment by ShadowJonathan** [comment]
You can try to rejoin those chats with `/join ROOMID SERVER`, replacing `ROOMID` with something like `!12345:server.com` and `SERVER` with `server.com`

Ask your friend for the room ID (room settings -> advanced -> internal room id), and use that as `ROOMID`, then put their server (the last part of their username, after `:`) as `SERVER`.

This should at least get you back into the rooms, but it's possible that the E2EE history is kinda lost, you'll have to ask around on `#synapse:matrix.org` on how to restore that.

If the chat room only existed on your own server, it's not really recoverable unless you can get more info out of your old server.

**Comment by wuast94** [comment]
On when I try "/join ROOMID:SERVER" I get unknown server

Of I make it any other way I get the massage "Was not legal room id..." 

Soi think that's not working :/

**Comment by ShadowJonathan** [comment]
you need to join with something like `/join !roomid:server.com`, not `/join roomid:server.com`

also, you need to add your friend's server after a space, so something like `/join !roomid:server.com otherserver.org`

**Comment by wuast94** [comment]
No matter what I do I only get the no legal room message :/ 

And even if it would work.. We won't get the 1:1 messages :/

**Comment by anoadragon453** [comment]
An "was not legal room ID or room alias" error will be returned if you've given the server an incorrectly-formatted identifier. Could you post a (possibly partially redacted) example of what you're typing? Or more preferably the exact network call that your client is sending to the server.

The latter can be found in your browser's (or Element Desktop's) developer tool's Network panel.

But to clarify, wiping your database does not have a supported recovery method outside of attempting to rejoin all of the rooms you were in. You can attempt to make this easier by asking other servers what rooms they currently see your server in, and then manually attempting to join those rooms. It's entirely possible that you may still run into issues down the line however - especially if you re-use the same Matrix ID.

**Comment by richvdh** [comment]
It's a bit unclear exactly what the complaint is here, but I think mostly it is a duplicate of https://github.com/matrix-org/synapse/issues/8487.
